### DISCUSSION TITLE
another use case

### DISCUSSION THREAD

**Discussion by cfritze64** [johnlindquist/kit#171] [General]
I'm the tech person for a chorus that's using some fancy sound tools to rehearse together in real time in a virtual rehearsal space. Then there are some singers who don't have the fancy tech and listen to the rehearsal via zoom meeting. I'm responsible for piping sound between these environments, which requires changing settings in the Mac's default output and input devices, changing Zoom's mic and speaker designations, and stitching other ports together via a command line tool. There are several scenarios required at different times in rehearsal - each with its different settings on these devices. Switching between them is a little hair-rising if done manually. Screw it up and 100+ participants are wondering what just screwed up as feedback blows their ears out. 

ScriptKit to the rescue! I'm developing a library of subroutines that let me set the various software switches via AppleScript or command line tools. And I'm increasingly abstracting the definitions for each environment. There are routines for setup and routines for execution, and eventually I'll use the tabs function in ScriptKit to develop a handy menu.

**Comment by johnlindquist** [comment]
Wow! Love this! Would love to see a screenshot/gif/vid of the final tool if you get it to a point you feel comfortable sharing. Good luck

**Reply by cfritze64** [reply]
[https://youtu.be/oFRzYhpr29A](https://youtu.be/oFRzYhpr29A)

first script connects to a jacktrip virtual sound room as a client.
based on parameters in a scenario definition object, the second script:
- changes the Mac's default in/out sound channels
- changes Zoom's mic and speaker channels
- builds an xml file that is passed to a jacktrip command line tool that patches particular channels to or from the jacktrip room

During a session we might switch between several scenarios, piping sound from zoom to jacktrip, or from jacktrip to zoom, and we might rely on a second jacktrip room if our group is in a zoom breakout room. The wiring is hard to memorize and a pain to switch manually, so it was a great help to abstract the process and write up some scripts...

**Reply by johnlindquist** [reply]
SO, SO, SO cool 😍

Thank you for sharing. That really makes my day!
